data-processor/src/main/java/io/micronaut/data/processor/visitors/finders/TypeUtils.java

```diff
@@ -360,6 +361,9 @@ public static Optional<DataType> resolveDataType(@NonNull ParameterElement param
         ClassElement genericType = parameter.getGenericType();
         Objects.requireNonNull(genericType);
         if (TypeUtils.isEntityContainerType(genericType) || genericType.hasStereotype(MappedEntity.class)) {
+            if (genericType.hasStereotype(JsonView.class)) {
```

Comment:
Please don't use JsonView

Reply:
I need to get JSON data type for binding. Maybe can annotate parameter in repository method visitor, will look at that option.

Reply:
Maybe you can decide later in the repository visitor based on the entity representation

Reply:
Changed it, looks better now and not using `@JsonView`.